rust/agents/relayer/src/msg/gas_payment/mod.rs

```diff
@@ -97,6 +97,17 @@ impl GasPaymentEnforcer {
                 ?whitelist,
                 "Message matched whitelist for policy"
             );
+
+            if current_payment.destination != message.destination {
```

Comment:
ok, this makes sense, the problem is it shouldn't panic on changing the destination domain for the same messageId in the`Add` impl of gas payments?

Reply:
Yeah exactly - that and because when we index gas payments we don't know exactly what destination is "correct", we should probably save all payments and separate them by the destination domain and then when we check the gas payment enforcement condition when we *do* have the correct destination domain, that's where the enforcement is